src/item/EndCrystal.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\item;
+
+use pocketmine\block\Block;
+use pocketmine\block\BlockTypeIds;
+use pocketmine\entity\Location;
+use pocketmine\entity\object\EnderCrystal;
+use pocketmine\math\AxisAlignedBB;
+use pocketmine\math\Facing;
+use pocketmine\math\Vector3;
+use pocketmine\player\Player;
+use function count;
+
+class EndCrystal extends Item{
+
+	public function onInteractBlock(Player $player, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector, array &$returnedItems) : ItemUseResult{
+		if($blockClicked->getTypeId() === BlockTypeIds::OBSIDIAN || $blockClicked->getTypeId() === BlockTypeIds::BEDROCK){
+			$pos = $blockClicked->getPosition();
+			$world = $pos->getWorld();
+			$bb = AxisAlignedBB::one()
+				->offset($pos->getX(), $pos->getY(), $pos->getZ())
+				->extend(Facing::UP, 1);
+			if(
+				count($world->getNearbyEntities($bb)) === 0 &&
```

Comment:
We can't put a crystal on a block where there are already items, so? Or is this code just to check that there are no crystals already there? Because if it checks for the presence of crystal, it checks more generally for the presence of any entity

Reply:
In bedrock vanilla you can't spawn crystal if there are any entities on top (such as players, projectiles, items, etc...)